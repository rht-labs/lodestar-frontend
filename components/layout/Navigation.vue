<template>
  <nav
    id="page-default-nav-primary-nav"
    class="pf-c-nav pf-m-dark"
    aria-label="Global"
  >
    <ul class="pf-c-nav__list">
      <li class="pf-c-nav__item">
        <nuxt-link class="pf-c-nav__link" to="/">Home</nuxt-link>
        <nuxt-link class="pf-c-nav__link" to="/one">Test Vuex Store</nuxt-link>
        <nuxt-link class="pf-c-nav__link" to="/two">Test Backend</nuxt-link>
      </li>
    </ul>
  </nav>
</template>

<script>
export default {}
</script>

<style scoped>
.pf-c-nav__link:focus {
  outline: none;
}
.pf-c-nav__link:active:focus {
  outline: none;
}
</style>
