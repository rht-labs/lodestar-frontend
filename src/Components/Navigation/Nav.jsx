import React from "react";
import {
  Nav,
  NavExpandable,
  NavItem,
  NavItemSeparator,
  NavList,
  NavGroup,
  NavVariants
} from "@patternfly/react-core";
import { Link } from "react-router-dom";

class NavDefaultList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      activeItem: 0
    };
    this.onSelect = result => {
      this.setState({
        activeItem: result.itemId
      });
    };
  }

  render() {
    const { activeItem } = this.state;
    return (
      <Nav onSelect={this.onSelect} theme="dark">
        <NavList>
          <NavItem
            id="engagementFormLink"
            itemId={0}
            isActive={activeItem === 0}
          >
            <Link to="/">Engagement Form</Link>
          </NavItem>
          <NavItem id="privateLink" itemId={3} isActive={activeItem === 3}>
            <Link to="/private">Private Page</Link>
          </NavItem>
          <NavItem id="secondPageLink" itemId={1} isActive={activeItem === 1}>
            <Link to="/one">Second Page</Link>
          </NavItem>
          <NavItem id="thirdPageLink" itemId={2} isActive={activeItem === 2}>
            <Link to="/two">Third Page</Link>
          </NavItem>
        </NavList>
      </Nav>
    );
  }
}

// const Logo = () => (
//   <ul>
//     <li>
//       <Link to="/">Form</Link>
//     </li>
//     <li>
//       <Link to="/about">About</Link>
//     </li>
//     <li>
//       <Link to="/dashboard">Dashboard</Link>
//     </li>
//   </ul>
// );

export default NavDefaultList;
