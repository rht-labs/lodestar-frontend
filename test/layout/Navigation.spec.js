import { shallowMount, createLocalVue } from '@vue/test-utils'
import Navigation from '@/components/layout/Navigation.vue'

const localVue = createLocalVue()

localVue.component('nuxt-link', {
  props: ['to'],
  template: '<a href="#"><slot>NuxtLink</slot></a>'
})

describe('Navigation', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(Navigation, {
      stubs: ['nuxt-link'],
      localVue
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = shallowMount(Navigation, {
      stubs: ['nuxt-link'],
      localVue
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
