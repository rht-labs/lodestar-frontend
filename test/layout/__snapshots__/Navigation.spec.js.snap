// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders properly 1`] = `"<nav id=\\"page-default-nav-primary-nav\\" aria-label=\\"Global\\" class=\\"pf-c-nav pf-m-dark\\"><ul class=\\"pf-c-nav__list\\"><li class=\\"pf-c-nav__item\\"><nuxt-link-stub to=\\"/\\" class=\\"pf-c-nav__link\\">Home</nuxt-link-stub> <nuxt-link-stub to=\\"/one\\" class=\\"pf-c-nav__link\\">Test Vuex Store</nuxt-link-stub> <nuxt-link-stub to=\\"/two\\" class=\\"pf-c-nav__link\\">Test Backend</nuxt-link-stub></li></ul></nav>"`;
